lib/plugins/aws/invokeLocal/index.js

```diff
@@ -148,9 +164,166 @@ class AwsInvokeLocal {
 
     const configuredEnvVars = this.getConfiguredEnvVars();
 
-    _.merge(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+    const promises = _.entries(configuredEnvVars).map(([name, value]) => {
+      return this.resolveResource(value).then(output => {
+        configuredEnvVars[name] = output;
+      });
+    });
+
+    return BbPromise.all(promises).then(() => {
+      Object.assign(process.env, lambdaDefaultEnvVars, credentialEnvVars, configuredEnvVars);
+    });
+  }
+
+  loadExportsImpl(token, cfExports) {
+    if (!cfExports) {
+      cfExports = [];
+    }
+    return this.provider
+      .request('CloudFormation', 'listExports', { NextToken: token })
+      .then(result => {
+        cfExports.push(...result.Exports);
+        if (result.NextToken) {
+          return this.loadExports(result.NextToken, cfExports);
```

Comment:
This function should be configured in a way, that only top level call is memoized, and internal invocation doesn't hit memoized function.

Ideally we should have method which takes no arguments, and which internally relies on helper which supports optional `NextToken` argument

Reply:
I've added a pass through function which is memoized.